web/packages/teleport/src/Users/UserAddEdit/TraitsEditor.test.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { fireEvent, render, screen } from 'design/utils/testing';
+
+import Validation from 'shared/components/Validation';
+
+import { AllUserTraits } from 'teleport/services/user';
+
+import { TraitsEditor, traitsToTraitsOption, emptyTrait } from './TraitsEditor';
+
+import type { TraitsOption } from './TraitsEditor';
+
+describe('Render traits correctly', () => {
+  const userTraits = {
+    logins: ['root', 'ubuntu'],
+    db_roles: ['dbadmin', 'postgres'],
+    db_names: ['postgres', 'aurora'],
+  } as AllUserTraits;
+
+  test('Available traits are rendered', async () => {
+    const configuredTraits = [] as TraitsOption[];
+    const setConfiguredTraits = jest.fn();
+
+    const { rerender } = render(
+      <TraitsEditor
+        allTraits={userTraits}
+        attempt={{ status: '' }}
+        configuredTraits={configuredTraits}
+        setConfiguredTraits={setConfiguredTraits}
+      />
+    );
+
+    expect(screen.getByText('User Traits')).toBeInTheDocument();
+    expect(setConfiguredTraits).toHaveBeenLastCalledWith(
+      traitsToTraitsOption(userTraits)
+    );
+
+    rerender(
+      <Validation>
+        <TraitsEditor
+          allTraits={userTraits}
+          attempt={{ status: '' }}
+          configuredTraits={traitsToTraitsOption(userTraits)}
+          setConfiguredTraits={setConfiguredTraits}
+        />
+      </Validation>
+    );
+    expect(screen.getAllByTestId('trait-key')).toHaveLength(3);
+    expect(screen.getAllByTestId('trait-value')).toHaveLength(3);
+  });
+
+  test('Add and remove Trait', async () => {
+    const userTraits = {} as AllUserTraits;
+    const configuredTraits = [] as TraitsOption[];
+    const setConfiguredTraits = jest.fn();
+
+    const { rerender } = render(
+      <Validation>
+        <TraitsEditor
+          allTraits={userTraits}
+          attempt={{ status: '' }}
+          configuredTraits={configuredTraits}
+          setConfiguredTraits={setConfiguredTraits}
+        />
+      </Validation>
+    );
+    expect(screen.queryAllByTestId('trait-key')).toHaveLength(0);
+
+    const addButtonEl = screen.getByRole('button', { name: /Add user trait/i });
+    expect(addButtonEl).toBeInTheDocument();
+    fireEvent.click(addButtonEl);
+
+    expect(setConfiguredTraits).toHaveBeenLastCalledWith([emptyTrait]);
+
+    const singleTrait = { logins: ['root', 'ubuntu'] };
+    rerender(
+      <Validation>
+        <TraitsEditor
+          allTraits={singleTrait}
+          attempt={{ status: '' }}
+          configuredTraits={traitsToTraitsOption(singleTrait)}
+          setConfiguredTraits={setConfiguredTraits}
+        />
+      </Validation>
+    );
+    fireEvent.click(screen.getByTitle('Remove Trait'));
+    expect(setConfiguredTraits).toHaveBeenCalledWith([]);
```

Comment:
This test block is not needed now as the  test case in first render is now no longer required since the `traitsToTraitsOption` is moved to  `useDialog.ts` now in https://github.com/gravitational/teleport/pull/42620/commits/314783e29151dc5c6c3035f684928298f6bf7000#diff-3e716db9176ae7ea0cef1294cddedf5c09b3d6f68b10e27ce6ecf676b49e9fcd.